src/settings.c

```diff
@@ -97,254 +97,301 @@ void init_default_settings()
     if (!silent) LOG_ERROR(("Parsing error: " msg ", \"%s\" found\n", str));
 
 /* Parse highlight color */
-int parse_scrollbars_highlight_color(char *str, char ***tgt, int silent)
+int parse_scrollbars_highlight_color(int argc, const char **argv, void **references, int silent)
 {
-    if (!strcasecmp(str, "disable"))
-        **tgt = NULL;
-    else {
-        **tgt = strdup(str);
-        if (**tgt == NULL) DIE_OOM(("Could not copy value from parameter\n"));
-    }
+    char **highlight_color = (char **) references[0];
```

Comment:
(Random spot, but applies to most of the changes in this PR): I think the new parsing functions much check that `argc == 1` unless they accept multiple arguments.

Reply:
I thought of validating the number of arguments in the parsing functions as well, especially since the responsibility of parameter validation is more often falls on the function than its callers, but [I embedded the allowed range of argument counts in the `Param` struct.](https://github.com/kolbusa/stalonetray/pull/27/files#diff-b16b9fd7ad76f07b54dd144e7c9917d017b82ab74ee7cd0cba9f7c750e2c509aR508-R509) That range is used for validation [before passing them to the parsing function,](https://github.com/kolbusa/stalonetray/pull/27/files#diff-b16b9fd7ad76f07b54dd144e7c9917d017b82ab74ee7cd0cba9f7c750e2c509aR1546-R1551) so I found it acceptable to delegate that to whatever procedure is reading `struct Param` type values.

Doing otherwise might simplify `void parse_rc()`, but in my opinion will lead to plenty of repetitive checks in each `parse_*` function. Additionally, we would either have to get rid of `int min_argc` and `int max_argc` in the structured type, relying on hardcoded values in each parsing function, or change the parser signature again to give it a `struct Param *` parameter and repeat the check in each parsing function:

```c
int parse_example(int argc, char **argv, struct Param *param, int silent)
{
  /* variable definitions */
  
  if (argc > param->max_argc || argc < param->min_argc) {
    PARSING_ERROR(/* ... */);
    return FAILURE;
  }

  /* ... */
}
```

What do you think is the best way forward? I agree that lack of validation in the parsing functions seems off, but I thought it was convenient to have as much information as possible embedded into the parameter type structure.

To sum up the options I see, none of which I find optimal:

1. Leave as-is, delegating responsibility to `parse_rc()` and `parse_cmdline()`.
2. Remove validation from `parse_rc()` and `parse_cmdline()`, remove argument count constraints from `struct Param`, hardcode valid range when validating `argc` in each parsing function.
3. Remove validation from `parse_rc()` and `parse_cmdline()` and move it into each parsing function, which will now receive a `struct Param *` argument.